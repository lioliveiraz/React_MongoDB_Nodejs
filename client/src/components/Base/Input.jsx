import React from "react";
import TextField from "@material-ui/core/TextField";
import { useStyles } from "./../../assets/css/Base/input";
import { capitalize } from "./../../helpers/services";

function Input(props) {
  const { type, name, placeholder, required, getUserInput, error } = props;
  const classes = useStyles();

  return (
    <>
      <TextField
        inputProps={{
          "data-cy": `input-${name}`,
        }}
        id="standard-secondary"
        color="secondary"
        variant="outlined"
        type={type}
        name={name}
        label={capitalize(name)}
        placeholder={placeholder}
        required={required}
        onInput={(e) => getUserInput(e.target.value, name)}
        className={classes.root}
        data-cy={`input-${name}-wrapper`}
      />
      {error && <p>{error}</p>}
    </>
  );
}

export default Input;
