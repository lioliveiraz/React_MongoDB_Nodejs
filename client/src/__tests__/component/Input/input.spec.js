import React from "react";
import { mount } from "enzyme";
import Input from "../../../components/Base/Input";

describe("Input", () => {
  let wrapper;
  const historyMock = { push: jest.fn() };

  afterEach(() => {
    wrapper.unmount();
  });

  it("should render correctly", () => {
    wrapper = mount(
      <Input
        type="text"
        name="name"
        placeholder="John"
        required={true}
        getUserInput={jest.fn()}
        error="Not valid"
      />
    );
    expect(wrapper).toMatchSnapshot();
    const label = wrapper.find("label");
    const errorMessage = wrapper.find("p");

    expect(label.text()).toEqual("Nameâ€‰*");
    expect(errorMessage.text()).toEqual("Not valid");
  });

  it("should not render error message", () => {
    wrapper = mount(
      <Input
        type="text"
        name="name"
        placeholder="John"
        required={true}
        getUserInput={jest.fn()}
      />
    );
    const errorMessage = wrapper.get("p");
    expect(errorMessage).toBeUndefined();
  });

  it("Should send right data", () => {
    let inputValue = {};

    wrapper = mount(
      <Input
        type="text"
        name="name"
        placeholder="John"
        required={true}
        getUserInput={(value, name) => (inputValue[name] = value)}
      />
    );
    const input = wrapper.find("[data-cy='input-name']");
    input.simulate("input", { target: { name: "name", value: "Ligia" } });
    expect(inputValue).toMatchObject({ name: "Ligia" });
  });
});
