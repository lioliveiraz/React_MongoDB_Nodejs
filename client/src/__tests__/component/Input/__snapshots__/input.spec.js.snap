// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `ReactWrapper {}`;

exports[`Input should match snapshot 2`] = `"<div class=\\"MuiFormControl-root MuiTextField-root makeStyles-root-1\\" data-cy=\\"input-name-wrapper\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"standard-secondary\\" id=\\"standard-secondary-label\\">Name<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\">â€‰*</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" id=\\"standard-secondary\\" name=\\"name\\" placeholder=\\"John\\" required=\\"\\" type=\\"text\\" data-cy=\\"input-name\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-4\\"><span>Name&nbsp;*</span></legend></fieldset></div></div>"`;
